operations/app/src/modules/function_app/main.tf

```diff
@@ -96,12 +96,12 @@ resource "azurerm_function_app" "function_app" {
     "APPINSIGHTS_INSTRUMENTATIONKEY" = var.ai_instrumentation_key
 
     # Test/Staging-specific app settings
-    "PRIME__HL7__USER" = (var.environment != "prod" ? var.az_phd_user : null)
-    "PRIME__HL7__PASS" = (var.environment != "prod" ? var.az_phd_password : null)
-    "PRIME__HL7_BATCH__USER" = (var.environment != "prod" ? var.az_phd_user : null)
-    "PRIME__HL7_BATCH__PASS" = (var.environment != "prod" ? var.az_phd_password : null)
-    "PRIME__CSV__USER" = (var.environment != "prod" ? var.az_phd_user : null)
-    "PRIME__CSV__PASS" = (var.environment != "prod" ? var.az_phd_password : null)
+    "PRIME__HL7__USER" = (var.environment != "prod" ? "@Microsoft.KeyVault(VaultName=${var.resource_prefix}-appconfig;SecretName=functionapp-az-phd-user)" : null)
+    "PRIME__HL7__PASS" = (var.environment != "prod" ? "@Microsoft.KeyVault(VaultName=${var.resource_prefix}-appconfig;SecretName=functionapp-az-phd-pass)" : null)
+    "PRIME__HL7_BATCH__USER" = (var.environment != "prod" ? "@Microsoft.KeyVault(VaultName=${var.resource_prefix}-appconfig;SecretName=functionapp-az-phd-user)" : null)
+    "PRIME__HL7_BATCH__PASS" = (var.environment != "prod" ? "@Microsoft.KeyVault(VaultName=${var.resource_prefix}-appconfig;SecretName=functionapp-az-phd-pass)" : null)
+    "PRIME__CSV__USER" = (var.environment != "prod" ? "@Microsoft.KeyVault(VaultName=${var.resource_prefix}-appconfig;SecretName=functionapp-az-phd-user)" : null)
+    "PRIME__CSV__PASS" = (var.environment != "prod" ? "@Microsoft.KeyVault(VaultName=${var.resource_prefix}-appconfig;SecretName=functionapp-az-phd-pass)" : null)
```

Comment:
Is this syntax required for either `app_settings` or because it is within a conditional?  Just curious since the `database` module accesses secrets as data sources.

Reply:
This is the syntax for a function's configuration to pull it's values from the Key Vault directly. `OKTA_clientId` has a cleaner example without the conditional.

We could follow the method used for the database user/pass, but technically it's less secure as the decrypted values have to pulled down into the Terraform state. Use the `@Microsoft.KeyVault` syntax for the function app removes a hop from the process.

Reply:
Ok, that makes sense.  Have we given the function app read access to secrets?